diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +119,200 @@ class LineLength : Rule("line-length") {
+            if (depthFirstSearch(firstParent, secondNode)!= null)
+                return firstParent
+            firstParent = firstParent.treeParent
+        }
+        return firstParent ?: firstNode
+    }
+
+    private fun depthFirstSearch(node: ASTNode, finder: ASTNode): ASTNode?{
+        if (node.getChildren(null).find { it == finder } != null)
+            return node
+        node.getChildren(null).forEach {
+            val answer = depthFirstSearch(it, finder)
+            if (answer != null) return answer
+        }
+        return null
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        when {
+            node.hasChildOfType(BINARY_EXPRESSION) -> {
+                node.getAllChildrenWithType(BINARY_EXPRESSION).forEach {
+                    searchBinaryExpression(it, binList)
+                }
+            }
+            node.hasChildOfType(PARENTHESIZED) -> {
+                node.getAllChildrenWithType(PARENTHESIZED).forEach {
+                    searchBinaryExpression(it, binList)
+                }
+            }
+            else -> {
+                binList.add(node)
+                binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+            }
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixProperty(parent: ASTNode, configuration: LineLengthConfiguration){
+        if (!parent.hasChildOfType(STRING_TEMPLATE)) {
+            if (parent.hasChildOfType(BINARY_EXPRESSION)) {
+                val leftOffset = positionByOffset(parent.findChildByType(BINARY_EXPRESSION)!!.psi.textOffset).second
+                fixCondition(parent, configuration, leftOffset, true)
+            }
+            if (parent.hasChildOfType(PARENTHESIZED)){
+                val newParent = parent.findChildByType(PARENTHESIZED)!!
+                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.psi.textOffset).second
+                fixCondition(newParent, configuration, leftOffset)
+            }
+        }
+        else
+            createNodes(parent, configuration)
+    }
+
+    private fun createNodes(parent: ASTNode, configuration: LineLengthConfiguration) {
+        var text = parent.findChildByType(STRING_TEMPLATE)!!.text
+        text = text.trimStart(text.first()).trimEnd(text.last())
```

Comment:
What will be trimmed here? Could you please add comment in code explaining why it is needed (something related to how strings are represented in the tree?)?